metron-platform/metron-parsers/src/test/java/org/apache/metron/writers/integration/WriterBoltIntegrationTest.java

```diff
@@ -166,6 +185,139 @@ public void test() throws UnableToStartException, IOException, ParseException {
     }
   }
 
+  /**
+   * {
+   *    "parserClassName":"org.apache.metron.writers.integration.WriterBoltIntegrationTest$EmptyObjectParser",
+   *    "sensorTopic":"emptyobjectparser",
+   *    "outputTopic": "enrichments",
+   *    "errorTopic": "parser_error"
+   * }
+   */
+  @Multiline
+  public static String offsetParserConfigJSON;
+
+  @Test
+  public void commits_kafka_offsets_for_emtpy_objects() throws Exception {
```

Comment:
Are `methods_named_like_this` acceptable according to our style guidelines?  If the Google Java guidelines say yes, then I am ok with it.  Otherwise, can we camel hump these names?

Reply:
My thoughts here probably belong in a discuss thread, but the example serves well in the moment. Just for disclosure, I almost always format my code using the IDE formatting tool that Justin shared some time ago with checkstyle. I'm inclined to want to break convention when the outcome can be of greater benefit than adhering strictly to the rules. The underscore pattern happens to be a test method naming convention that a group of 30 engineers and myself landed on 7 years ago while re-writing a million+ line application that had no tests from the ground up to include unit, integration, and acceptance tests.  After a lot of trial and error, it was the most clear for everyone involved.

Which do you prefer to read? Which reads most naturally? Which provides context about the test without needing extraneous javadoc?

```
commits_kafka_offsets_for_emtpy_objects()
CommitsKafkaOffsetsForEmtpyObjects()
testCommits()
test()
```

**TL;DR**

Per the Google style guide, this naming convention is allowable:
> Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in lowerCamelCase. One typical pattern is <methodUnderTest>_<state>, for example pop_emptyStack. **There is no One Correct Way to name test methods.**

https://google.github.io/styleguide/javaguide.html#s5.2-specific-identifier-names
https://stackoverflow.com/questions/96297/what-are-some-popular-naming-conventions-for-unit-tests
https://dzone.com/articles/7-popular-unit-test-naming

Reply:
I vote for one that doesn't contain "emtpy" in it.

Reply:
No doubt I've seen it done in other code bases.  Its just not common in the Metron code base, thus I called it out.  

Since it doesn't violate the code style guidelines, then I have no basis to complain.  I wasn't aware that was acceptable.  Thanks for clarifying.

Reply:
> @justinleet I vote for one that doesn't contain "emtpy" in it.

And "emtpy" is actually the Olde English spelling, which I am fine with.

Reply:
I agree @nickwallen - it isn't the prevailing idiom that we've used. And regardless of whatever the current standards are, I'm a big fan of open dialogue and comments like the ones you made here.